lib/manager/cargo/extract.ts

```diff
@@ -63,13 +85,50 @@ function extractFromSection(
 
 export function extractPackageFile(
   content: string,
-  fileName: string
+  fileName: string,
+  config?: ExtractConfig
 ): PackageFile | null {
   logger.trace(`cargo.extractPackageFile(${fileName})`);
-  let parsedContent: CargoConfig;
+
+  let cargoConfig: CargoConfig | undefined;
+  if (config?.localDir) {
```

Comment:
This condition is not necessary.

Reply:
In tests, `config` is not passed - and `localDir` might always be set, but it's optional as far as TypeScript is concerned:

```typescript
export interface ManagerConfig {
  binarySource?: string;
  dockerUser?: string;
  localDir?: string;
  registryUrls?: string[];
}
```